hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdb/util/ManagedSstFileReader.java

```diff
@@ -89,16 +90,31 @@ public long getEstimatedTotalKeys() throws RocksDBException {
   }
 
   public Stream<String> getKeyStream() throws RocksDBException {
+    return getKeyStream(Optional.empty(), Optional.empty());
+  }
+
+  public Stream<String> getKeyStream(Optional<String> lowerBound,
+      Optional<String> upperBound) throws RocksDBException {
     // TODO: [SNAPSHOT] Check if default Options and ReadOptions is enough.
     final MultipleSstFileIterator<String> itr =
         new MultipleSstFileIterator<String>(sstFiles) {
           private ManagedOptions options;
           private ReadOptions readOptions;
 
+          private Optional<ManagedSlice> lowerBoundSLice;
+
+          private Optional<ManagedSlice> upperBoundSlice;
```

Comment:
Do you need to close them in [close](https://github.com/apache/ozone/blob/13da8647d368b5d2a17857d3535d72c772801f3e/hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdb/util/ManagedSstFileReader.java#L133)? If yes, please do so otherwise make them local variable to the `init` method.

Reply:
we need to close all rocks object manually. As close actually triggers the delete for the dangling pointers on the c++ side.

Reply:
Then close is missing.

Reply:
done